### PR TITLE
Add Rivets implementation + CR changes from #1226 + merged

### PR DESCRIPTION
**Pull request by altmind**
This is full and exact duplicate of @mikeric implementation in #1226 with following requested changes:
* removed empty line in html
* added autofocus on double click on todo
* resolved merge conflict(deliberately make this separate commit to distinguish new code and merge changes)

New PR as I'm pushing from my personal fork

### CODE REVIEW

**Comment by samccone** [comment]
jscs is :white_check_mark: 
integration suite needs to be run manually :warning:

**Comment by arthurvr** [comment]
Can we update this to todomvc-app-css@1.0.4 and the latest todomvc-common?

**Comment by altmind** [comment]
@arthurvr done, please review.

**Comment by altmind** [comment]
Ping.

**Comment by samccone** [comment]
@altmind mind squashing this into a single commit?

**Comment by arthurvr** [comment]
@altmind You shouldn't start doing ping comments after one day :) We'll be happy to review this but we're all busy :)

Anyways, thanks for working on this! Two other things:

* [ ] The clear completed button is doing weird things. It shows ` Clear completed ()` when there are no todos. Take a look at the app spec, we anyways don't require that count anymore.
* [ ] I see 2 failing specs locally. Those sound related to the above, though.

File: examples/rivets/readme.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Learning Rivets
+
+The [Rivets homepage](http://rivetsjs.com) is a great resource for getting started.
+
+Here are some links you may find helpful:
+
+* [Guide](http://rivetsjs.com/docs/guide/)
+* [Binder Reference](http://rivetsjs.com/docs/reference/)
+* [Rivets on Stack Overflow](http://stackoverflow.com/questions/tagged/rivets.js)
```
Comment (arthurvr) [inline examples/rivets/readme.md]: Stack overflow and linking up the official site twice is okay but isn't there anything else we can bring up in here? Interesting blogposts maybe?

File: examples/rivets/index.html [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+<html lang="en" data-framework="rivets">
+	<head>
+		<meta charset="utf-8">
+		<title>Rivets • TodoMVC</title>
+		<link rel="stylesheet" href="node_modules/todomvc-common/base.css">
+		<link rel="stylesheet" href="node_modules/todomvc-app-css/index.css">
+	</head>
+	<body>
+		<script type="text/template" id="todos">
+			<header id="header">
```
Comment (samccone) [inline examples/rivets/index.html]: we should use the 2.0 spec with classes and not IDs

Comment (arthurvr) [inline examples/rivets/index.html]: yeah. The tests aren't updated but let's use todomvc-app-css 2.0 with classes. We'll update the tests soon anyways.

Comment (samccone) [inline examples/rivets/index.html]: yep I was thinking about the work last night required for a dynamic suite, should actually be quite simple :cocktail:

Comment (altmind) [inline examples/rivets/index.html]: guys, i dont know what are you talking about :)

Comment (arthurvr) [inline examples/rivets/index.html]: https://github.com/tastejs/todomvc-app-css/issues/4 =)

Comment (samccone) [inline examples/rivets/index.html]: https://github.com/tastejs/todomvc/pull/1304/files#diff-34eab039025c337bc6534509702dbb78R18

see how we use classes here?

and the 2.0 app css
https://github.com/tastejs/todomvc/pull/1304/files#diff-e6462df4cdf27b57d669e5c8a2dba09fR18

:) that would be ideal for this implementation

Comment (altmind) [inline examples/rivets/index.html]: * Fixed `Clear completed ()`
* Fixed test, output: https://gist.github.com/altmind/614b15cb5f16d41d4707 It seems that tests/README.md is out of sync. used `mocha allTests` directly

Comment (altmind) [inline examples/rivets/index.html]: @samccone can you please reassure me that i need to use classes and css@2.0.
You wont got any test coverage - all the tests fail.

Comment (samccone) [inline examples/rivets/index.html]: I reassure you :)

The tests will only fail until I fix them :)

Comment (altmind) [inline examples/rivets/index.html]: bam! done.

**Comment by altmind** [comment]
Can this be merged already? I addressed and fixed all the comments. It takes weeks just to merge working implementation :(

**Comment by sindresorhus** [comment]
We don't get notifications on new commits to a pull request. And please stop the complaining. This is Open Source and we're all doing this in our free time. By opening a pull request you're implicitly asking us to do work (yes, reviewing a pull request takes a lot of work). We'll try to get this merged soon.

**Comment by altmind** [comment]
I just want things to connect and something better to happen. For common benefit. I care.

I am struggling to create implementation for 3 months already  https://github.com/tastejs/todomvc/pull/1176

And it always a little bit that is missing. And dangling PRs are slowly decaying(as new requirements require PR changes).

And its demoralizing that i cannot do anything else to help :(

**Comment by arthurvr** [comment]
> I am struggling to create implementation for 3 months already

Please do note, though, that's mainly because at https://github.com/tastejs/todomvc/pull/1226 the author got inactive and we needed to close it due to inactivity. Anyways, let's move on, I will review all the code again (hopefully this evening) but I'm pretty hopeful that we'll be able to land this soon.

File: examples/rivets/js/stores/todo-store.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		// Removes a todo and saves.
+		remove: function (todo) {
+			this.all.splice(this.all.indexOf(todo), 1);
+			this.save();
+		},
+
+		// Toggles the todo as complete / active and saves.
+		toggle: function (todo) {
+			todo.completed = !todo.completed;
+			this.save(true);
```
Comment (samccone) [inline examples/rivets/js/stores/todo-store.js]: as @sindresorhus pointed out last week to me

http://ariya.ofilabs.com/2011/08/hall-of-api-shame-boolean-trap.html

File: examples/rivets/package.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "private": true,
+  "dependencies": {
+    "todomvc-common": "^1.0.2",
+    "todomvc-app-css": "^2.0.0",
+    "director": "^1.2.0",
+    "sightglass": "^0.2.4",
```
Comment (samccone) [inline examples/rivets/package.json]: what is this adding?

**Comment by arthurvr** [comment]
What Sam said. For the rest the code looks really good! Thanks for your hard work on this one!

**Comment by altmind** [comment]
Fixed https://github.com/tastejs/todomvc/pull/1295#discussion_r31052270 and https://github.com/tastejs/todomvc/pull/1295#discussion_r31052315

Seems like thats it.

File: examples/rivets/.gitignore [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+!node_modules/todomvc-common/base.js
+
+node_modules/todomvc-app-css/*
+!node_modules/todomvc-app-css/index.css
+
+node_modules/director/*
+!node_modules/director/build/director.js
+
+node_modules/rivets/*
+!node_modules/rivets/dist/rivets.bundled.min.js
```
Comment (arthurvr) [inline examples/rivets/.gitignore]: The gitignore looks good but this isn't actually committed.

File: examples/rivets/js/stores/todo-store.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+		// Saves a snapshot of all todos in local storage.
+		save: function (opts) {
+			if (opts && opts.notifyChanges) {
+				this.all.splice(0, 0);
+			}
+
+			localStorage['todos-rivets'] = JSON.stringify(this.all);
+		}
+	}
```
Comment (arthurvr) [inline examples/rivets/js/stores/todo-store.js]: missing semicolon. Please run jshint next time :)

Comment (altmind) [inline examples/rivets/js/stores/todo-store.js]: @arthurvr strange, both IDEA and jscs show no jscs errors (using .jscsrc from repo root)
```
D:\docs\devel\todomvc>jscs -c .jscsrc D:\docs\devel\todomvc\examples\rivets\js\stores\todo-store.js
No code style errors found.
```

Should I do anything about it?

Comment (arthurvr) [inline examples/rivets/js/stores/todo-store.js]: Also run JSHint ;)

Comment (altmind) [inline examples/rivets/js/stores/todo-store.js]: @arthurvr oops. added semicolon :)

**Comment by arthurvr** [comment]
@altmind I will test some more this evening but the code looks good to me. Thanks for the updates.

**Comment by arthurvr** [comment]
:+1: Going to give people some time to complain before landing but this LGTM.

**Comment by samccone** [comment]
Hi @altmind I just tested this app for memory leaks and it looks like this has a few. :fountain: 

-----

##### Steps  to reproduce

* start timeline profiler
  * add three todos
  * delete three todos
  * (repeat above 2 more times)
* force GC
* stop timeline profiler

screen shot 2015-06-07 at 12 59 16 pm

screen shot 2015-06-07 at 12 59 59 pm


----

These should be addressed before we merge

**Comment by arthurvr** [comment]
Nice catch @samccone. Ping @altmind about the above.

**Comment by arthurvr** [comment]
I'm going to close this due to inactivity. Happy to reopen when you're willing to finish things up. This is sadly the 3th for the rivets example which we need to close due to inactivity.

> Can this be merged already? I addressed and fixed all the comments. It takes weeks just to merge working implementation :(

Now you can see yourself why ;)

**Pull request closed without merging**